core/src/main/java/org/arquillian/smart/testing/configuration/ConfigurationLoader.java

```diff
@@ -22,26 +22,35 @@
 
     public static final String SMART_TESTING_YML = "smart-testing.yml";
     public static final String SMART_TESTING_YAML = "smart-testing.yaml";
+    public static final String SMART_TESTING_CONFIG = "smart.testing.config";
+
     private static final Logger logger = Log.getLogger();
 
     public static Configuration load() {
         return load(Paths.get("").toAbsolutePath().toFile());
     }
 
     public static Configuration load(File projectDir) {
-        final File[] files =
-            projectDir.listFiles((dir, name) -> name.equals(SMART_TESTING_YML) || name.equals(SMART_TESTING_YAML));
-
-        if (files == null) {
-            throw new RuntimeException("I/O errors occurs while listing dir " + projectDir);
-        }
-
+        String customConfigFile = System.getProperty(SMART_TESTING_CONFIG);
         Map<String, Object> yamlConfiguration = new HashMap<>(0);
-        if (files.length == 0) {
-            logger.info("Config file `" + SMART_TESTING_YAML + "` OR `" + SMART_TESTING_YML + "` is not found. "
-                + "Using system properties to load configuration for smart testing.");
+
+        if (customConfigFile != null) {
+            File file = Paths.get(customConfigFile).toAbsolutePath().toFile();
```

Comment:
I'd add some protection of kind if file does not exists then log as warning explaining that the specified file cannot be found. Like it is done in else block `files.length()==0`